src/pages/Products/Products.tsx

```diff
@@ -1,44 +1,123 @@
-import { useEffect } from "react";
+import { useCallback, useEffect, useState } from "react";
 import { AppDispatch } from "../../redux/app/store";
 import { useDispatch } from "react-redux";
 import { useAppSelector } from "../../main/hooks";
 import { getAllProducts } from "../../redux/features/products/productsSlice";
 import { ProductItem } from "../../components/Product/ProductItem";
 import { AsideFilter } from "../../components/Filters/AsideFilter";
+import { AllProductsHeader } from "../../components/AllProductsHeader/AllProductsHeader";
+import { ControlContainer } from "../../components/ControlContainer/ControlContainer";
+import { Select } from "../../components/Select/Select";
+import { sortOptions } from "../../main/constants/filter.sort.data";
+import { PaginationController } from "../../components/Pagination/Pagination";
+import { Button } from "../../components/Button/Button";
+import { Bage } from "../../components/Bage/Bage";
+import { Product } from "../../redux/features/products/types";
 
 export const Products = () => {
   const dispatch: AppDispatch = useDispatch();
-  const { products } = useAppSelector((state) => state.product);
   const { selectedFilters } = useAppSelector((state) => state.filters);
+  const { products, total, limit, page } = useAppSelector(
+    (state) => state.product,
+  );
+  const totalPages = Math.ceil(total / limit);
+
+  const [sortData, setSortData] = useState<{ field: string; order: string }>();
+  const [currentPage, setCurrentPage] = useState<number>(page);
+  const [visibleProducts, setVisibleProducts] = useState<Product[]>([]);
+  const [showMoreMode, setShowMoreMode] = useState<boolean>(false);
+
+  const getSortParams = useCallback((value: string): void => {
+    const [field, order] = value.split("_");
+    setSortData({ field, order });
+  }, []);
 
   useEffect(() => {
     dispatch(
       getAllProducts({
-        page: 1,
-        limit: 5,
+        page: currentPage,
+        limit,
         brands: selectedFilters.brands,
         priceMax: selectedFilters.price.max,
         priceMin: selectedFilters.price.min,
         rating: selectedFilters.rating || 5,
+        sortField: sortData?.field,
+        sortDirection: sortData?.order,
       }),
     );
-  }, [selectedFilters]);
+  }, [selectedFilters, sortData, currentPage]);
+
+  useEffect(() => {
+    if (showMoreMode) {
+      setVisibleProducts((prev) => {
+        const uniqueProducts = new Map();
+        [...prev, ...products].forEach((product) => {
+          uniqueProducts.set(product._id, product);
+        });
+        return Array.from(uniqueProducts.values());
```

Comment:
Empty line before return. But I have a couple of questions about this part. Let`s speak tomorrow